### PR TITLE
Refactored the thumbs up and down so that it's one dry thumb.

### PR DESCRIPTION
**Pull request by roseak**
@rrgayhart I refactored my code to DRY it up.  I had two different functions for changing the up and down thumb qualities with the only difference being the thumbsUp or thumbsDownMap.  I pulled out the maps and passed them in as variables and then just made one thumbs function. 

strike

### CODE REVIEW

**Comment by rrgayhart** [comment]
Hey Rose! 

Two things to note:

Part of the assignment was using the PR template to explain the work that you did in the body of the PR: https://gist.github.com/rrgayhart/c64f0966a36a9c47b227
​
Not a big deal because I think you gave a good explanation here in the body, but wanted to make sure you had a look at it because it's a good way of thinking about how you explain work in a PR body to give other reviewers context.
​
Also, I'd like to see another student review this. You mentioned that your refactor buddy's were out, so feel free to tag someone else. Regis, Tyler, Jeff and Ricky have all been giving great feedback.

File: app/assets/javascripts/ideas.js [hunk 1]
```diff
@@ -12,6 +12,18 @@ function getIdeas(){
   });
 };
 
+var thumbsUpMap = {
```
Comment (rrgayhart) [inline app/assets/javascripts/ideas.js:15]: Nice!

File: db/schema.rb [hunk 2]
```diff
@@ -10,4 +24,12 @@
     t.datetime "updated_at",             null: false
   end
 
+  create_table "skills", force: :cascade do |t|
```
Comment (rrgayhart) [inline db/schema.rb]: Be careful about adding in files that don't directly relate to the PR into the PR code. Based on the changes I saw, this shouldn't be in here.

**Comment by rrgayhart** [comment]
This was a great way to refactor the code! And excellent .gif usage! 

bowling

**Comment by roseak** [comment]
Good point @rrgayhart!  I removed the random extra table.  Thanks for reviewing my code! @jbrr Hey, can you check out my awesome refactoring job?  You're the best.

File: app/assets/javascripts/ideas.js [hunk 3]
```diff
@@ -161,51 +174,20 @@ function updateBody(idea, body){
...
-}
-
 function updateQuality(idea, quality){
   $(idea).find('.quality').html('Quality: ' + quality);
   $(idea).attr('data-quality', quality);
 }
 
-function thumbsDown(){
-  $('#thumbs-down-idea').on('click', function(){
+function thumbed(direction, map){
```
Comment (jbrr) [inline app/assets/javascripts/ideas.js:182]: DRYing out your code, this is great!

**Comment by jbrr** [comment]
I like this refactor a lot, DRYing things out, and also going for the whole "Tell, Don't Ask" philosophy, in so far as now you're just telling the quality which direction it's going instead of making decisions based on it's state. Also, good catch on the extra table in the database, that kind of stuff is super easy to overlook.
Nice job! I'd say ship it!

**Pull request merged**